brutils/titulo_eleitoral.py

```diff
@@ -0,0 +1,167 @@
+        is a valid string conforming with the titulo eleitoral build rules
+    """
+
+    # ensure 'numero_titulo' only contains numerical characters
+    if not isinstance(numero_titulo, str) or not numero_titulo.isdigit():
+        return False
+
+    # split string into 'numero sequencial', 'unidade federativa' &
+    # 'digitos verificadores'
+    # edge case: 13-digit length mitigates here. The 9th sequential
+    # digit is not used for calculations.
+    tit_sequence, tit_uf, tit_dig_verifiers = _split_string(numero_titulo)
+
+    # verify length
+    if not _verify_length(numero_titulo, tit_uf):
+        return False
+    else:
+        lentgh_verified = True
+
+    # list valid UFs
+    unidades_federativas = ["{:02d}".format(i) for i in range(1, 29)]
+
+    # calculate dv1
+    dv1, verified_dv1 = _calculate_dv1(tit_sequence, tit_uf, tit_dig_verifiers)
+
+    # calculate dv2
+    verified_dv2 = _calculate_dv2(tit_uf, dv1, tit_dig_verifiers)
+
+    # verify UF
+    verified_uf = _verify_uf(tit_uf, unidades_federativas)
+
+    # return True if all conditions are met, else False
+    return all([verified_dv1, verified_dv2, lentgh_verified, verified_uf])
+
+
+def _split_string(input_string: str):
+    """split string into 'numero sequencial', 'unidade federativa' &
+    'digitos verificadores'
+    here I indexed tit_uf and tit_dig_verifiers from the back due to the
+    below reference:
+    '- Alguns títulos eleitorais de São Paulo e Minas Gerais podem ter nove
+    dígitos em seu número sequencial, em vez de oito. Tal fato não compromete
+    o cálculo dos DVs, pois este é feito com base nos oito primeiros números
+    sequenciais.'
+    http://clubes.obmep.org.br/blog/a-matematica-nos-documentos-titulo-de-eleitor/
+
+    Args:
+            input_string[str]: an example of titulo eleitoral brasileiro
+
+    Returns:
+        tit_sequence[list]: sequential number sliced from input_string
+        tit_uf[list]: Unidade Federal sliced from input_string
+        tit_dig_verifiers[list]: DV sliced from input_string
+    """
+    tit_sequence = input_string[:8]
+    tit_uf = input_string[-4:-2]
+    tit_dig_verifiers = input_string[-2:]
+
+    return tit_sequence, tit_uf, tit_dig_verifiers
+
+
+def _verify_length(numero_titulo, tit_uf):
+    """
+    verify length of numero_titulo
+    considers edge case with 13 digits for SP & MG
+    Args:
+        numero_titulo[str]: an example of titulo eleitoral brasileiro
+        tit_uf[list]: Unidade Federal sliced from input_string
+
+    Returns:
+        lentgh_verified[bool]: boolean indicating whether length of
+        numero_titulo is verified, or not.
+
+    """
+    if len(numero_titulo) == 12:
+        lentgh_verified = True
+
+    # edge case: for SP & MG with 9 digit long 'numero sequencial'
+    elif len(numero_titulo) == 13 and tit_uf in ["01", "02"]:
+        lentgh_verified = True
+    else:
+        lentgh_verified = False
+
+    return lentgh_verified
```

Comment:
```suggestion
    # edge case: for SP & MG with 9 digit long 'numero sequencial'
    if len(numero_titulo) == 13 and tit_uf not in ["01", "02"]:
        return False

    return len(numero_titulo) == 12
```

Reply:
neste caso, se o numero_titulo tiver 13 digitos e o uf for de SP ou MG deveria retornar True entao a alternativa nao segue a logica. Estou deixando algo passar?

Reply:
opa, foi mal, sugeri errado mesmo 🙈  queria ter sugerido:

```
    if len(numero_titulo) == 13 and tit_uf in ["01", "02"]:
        return True
```